lib/s3routes/routesUtils.ts

```diff
@@ -689,16 +718,15 @@ export function streamUserErrorPage(
     } else {
         error = errors.InternalError.customizeDescription(err.message);
     }
+    // Prepare the headers, but do not send them
+    // as errors might still happen when retrieving the data
     setCommonResponseHeaders(corsHeaders, response, log);
     response.setHeader('x-amz-error-code', error.message);
     response.setHeader('x-amz-error-message', error.description);
-    response.writeHead(error.code, { 'Content-type': 'text/html' });
-    response.on('finish', () => {
-        log.end().info('responded with streamed content', {
-            httpCode: response.statusCode,
-        });
-    });
-    return retrieveData(dataLocations, retrieveDataParams, response, log);
+    return retrieveData(dataLocations, retrieveDataParams, response, {
+        statusCode: error.code,
+        headers: { 'Content-type': 'text/html' },
```

Comment:
`'Content-type': 'text/html'` can be set with response.setHeader ?
(and since it is a header, it can be "overwritten" later if we actually send an error?)

Reply:
It was moved to a `setHeader`: it is basically the same at the end, but at least it simplifies how we pass the http code in the stream data function.